modules/troubleshooting-enabling-kdump-day-one.adoc

```diff
@@ -0,0 +1,82 @@
+* Node unavailability while the kernel is dumping the core.
+* Additional storage space being used to store the crash dumps.
+
+If you are aware of the downsides and trade-offs of having the `kdump` service enabled, it is possible to enable kdump in a cluster-wide fashion. Although machine-specific machine configs are not yet supported, you can use a `systemd` unit in a `MachineConfig` object on day-1 and have kdump enabled on all nodes in the cluster. You can create a `MachineConfig` object and inject that object into the set of manifest files used by Ignition during cluster setup.
+
+[NOTE]
+====
+See "Customizing nodes" in the _Installing -> Installation configuration_ section for more information and examples on how to use Ignition configs.
+====
+
+.Procedure
+
+Create a `MachineConfig` object for cluster-wide configuration:
+
+. Create a Butane config file, `99-worker-kdump.bu`, that configures and enables kdump:
++
+[source,yaml]
+----
+variant: openshift
+version: 4.11.0
+metadata:
+  name: 99-worker-kdump <1>
+  labels:
+    machineconfiguration.openshift.io/role: worker <1>
+openshift:
+  kernel_arguments: <2>
+    - crashkernel=256M
+storage:
+  files:
+    - path: /etc/kdump.conf <3>
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+          path /var/crash
+          core_collector makedumpfile -l --message-level 7 -d 31
+
+    - path: /etc/sysconfig/kdump <4>
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+          KDUMP_COMMANDLINE_REMOVE="hugepages hugepagesz slub_debug quiet log_buf_len swiotlb"
+          KDUMP_COMMANDLINE_APPEND="irqpoll nr_cpus=1 reset_devices cgroup_disable=memory mce=off numa=off udev.children-max=2 panic=10 rootflags=nofail acpi_no_memhotplug transparent_hugepage=never nokaslr novmcoredd hest_disable"
+          KEXEC_ARGS="-s"
+          KDUMP_IMG="vmlinuz"
+
+systemd:
+  units:
+    - name: kdump.service
+      enabled: true
+----
++
+<1> Replace `worker` with `master` in both locations when creating a `MachineConfig` object for control plane nodes.
+<2> Provide kernel arguments to reserve memory for the crash kernel. You can add other kernel arguments if necessary.
+<3> If you want to change the contents of `/etc/kdump.conf` from the default, include this section and modify the `inline` subsection accordingly.
+<4> If you want to change the contents of `/etc/sysconfig/kdump` from the default, include this section and modify the `inline` subsection accordingly.
+
+. Use Butane to generate a machine config YAML file, `99-worker-kdump.yaml`, containing the configuration to be delivered to the nodes:
++
+[source,terminal]
+----
+$ butane 99-worker-kdump.bu -o 99-worker-kdump.yaml
+----
+
+. Put the YAML file into manifests during cluster setup. You can also create this `MachineConfig` object after cluster setup with the YAML file:
++
+[source,terminal]
+----
+$ oc create -f ./99-worker-kdump.yaml
```

Comment:
`./` is spurious; It just means the current directory which will be the default anyway.